src/System.Linq.Expressions/src/System/Linq/Expressions/MemberExpression.cs

```diff
@@ -276,7 +280,10 @@ public static MemberExpression Property(Expression expression, PropertyInfo prop
         {
             ContractUtils.RequiresNotNull(property, "property");
 
-            MethodInfo mi = property.GetGetMethod(true) ?? property.GetSetMethod(true);
+            MethodInfo getter = property.GetGetMethod(true);
+            MethodInfo setter = property.GetSetMethod(true);
```

Comment:
Should scan for Set method only if Get is not available. Write-only properties are very rare.